#!/usr/bin/python3
=======================
Tests for add_integer
=======================

>>> import importlib
>>> add_integer = importlib.import_module('0-add_integer').add_integer

>>> add_integer(1, 2)
3

>>> add_integer(100, -2)
98

>>> add_integer(2)
100

>>> add_integer(100.3, -2)
98

>>> add_integer(None, 2)
Traceback (most recent call last):
    ...
TypeError: a must be an integer

>>> add_integer(2, None)
Traceback (most recent call last):
    ...
TypeError: b must be an integer

>>> add_integer(4, "School")
Traceback (most recent call last):
    ...
TypeError: b must be an integer

>>> add_integer(float('inf'), 2)
Traceback (most recent call last):
    ...
TypeError: a must be a valid number

>>> add_integer(2, float('-inf'))
Traceback (most recent call last):
    ...
TypeError: b must be a valid number

>>> add_integer(float('nan'), 2)
Traceback (most recent call last):
    ...
TypeError: a must be a valid number

>>> add_integer(2, float('nan'))
Traceback (most recent call last):
    ...
TypeError: b must be a valid number
